### DISCUSSION TITLE
What is the solution to this?

### DISCUSSION THREAD

**Discussion by Reyororo** [FunkinCrew/Funkin#1893] [Compiling help] [answered]
I ran lime test windows and got this error
"source/StoryMenuState.hx:44: characters 9-10 : Expected ]"The part of the letters 9-10 is the "," following the dad. What can I do about this?
![スクリーンショット 2021-11-28 132015](https://user-images.githubusercontent.com/88378158/143729538-7af0b749-c17e-418c-a940-669c7f8b1ee6.png)
![スクリーンショット 2021-11-28 131911](https://user-images.githubusercontent.com/88378158/143729542-dd85cf52-3f11-4566-9a83-e219e70e50a7.png)

**Comment by libauror-a** [comment, marked as answer]
Why are you yousing visual studio? the haxe language isn't supported in the program. Use something like Visual Studio Code with Haxe language server and Lime, also you forgot the conmma at the end of the `['senpai', 'bf', 'gf]` line

**Reply by Reyororo** [reply]
I've read that windows must be visual studio 2019, but is there any problem using something like Visual Studio Code with Haxe language server and Lime?

**Reply by Reyororo** [reply]
Again, an error has occurred
![2](https://user-images.githubusercontent.com/88378158/143841244-69348ef7-1c1c-455f-b193-81ba2b0a6adc.png)
![2](https://user-images.githubusercontent.com/88378158/143841263-9bdc2769-d3b7-474c-a5a0-57543fc88add.png)

**Reply by TheHENOOB** [reply]
> 
> I've read that windows must be visual studio 2019, but is there any problem using something like Visual Studio Code with Haxe language server and Lime?
> 

The thing is, to compile fnf you must have visual studio BUT not to write code, but to get the tools or compilers the README.MD is showing, you write code on Visual Studio Code using the haxe and lime extension, so you can see the error

**Reply by Reyororo** [reply]
> > I've read that windows must be visual studio 2019, but is there any problem using something like Visual Studio Code with Haxe language server and Lime?
> 
> The thing is, to compile fnf you must have visual studio BUT not to write code, but to get the tools or compilers the README.MD is showing, you write code on Visual Studio Code using the haxe and lime extension, so you can see the error


Can I use any visual studio that has the haxe extension?

**Reply by TheHENOOB** [reply]
> > > I've read that windows must be visual studio 2019, but is there any problem using something like Visual Studio Code with Haxe language server and Lime?
> > 
> > 
> > The thing is, to compile fnf you must have visual studio BUT not to write code, but to get the tools or compilers the README.MD is showing, you write code on Visual Studio Code using the haxe and lime extension, so you can see the error
> 
> Can I use any visual studio that has the haxe extension?

No, you only use VS Code because Visual Studio and VS Code extensions are different and is also the recommended way to write haxe code if you want to use a IDE

**Reply by Cheemsandfriends** [reply]
Guys...
There's HaxeDevelop

**Reply by libauror-a** [reply]
> Guys... There's HaxeDevelop

i literally forgot that the program existed

**Reply by TheHENOOB** [reply]
> Guys...
> There's HaxeDevelop

Bad option, it haven't been updated for years

I know that there's better editors but in case of Haxe, VS Code is the only modern available option when using a IDE or just the notepad if you don't plan on working with code in a mod to much since funkin has some available tools

**Reply by Cheemsandfriends** [reply]
> > Guys...
> > There's HaxeDevelop
> 
> Bad option, it haven't been updated for years
> 
> I know that there's better editors but in case of Haxe, VS Code is the only modern available option when using a IDE or just the notepad if you don't plan on working with code in a mod to much since funkin has some available tools

Well, Yeah But I'm just being kinda passive on what IDE should use cos that or that.
Just let ppl use what they want lol

**Reply by Reyororo** [reply]
I'd like to thank you all for solving this problem for me, and I understand it now ;)

**Comment by Cheemsandfriends** [comment]
You forgot a Comma in the senpai array.
The comma is like for marking the next thing
so like 
```
                        ['dad', 'bf', 'gf'],
			['dad', 'bf', 'gf'],
			['spooky', 'bf', 'gf'],
			['pico', 'bf', 'gf'],
			['mom', 'bf', 'gf'],
			['parents-christmas', 'bf', 'gf'],
			['senpai', 'bf', 'gf'],
                        ['dad', 'bf', 'gf']
```
Also, Do not place a comma if you're not continuing the array.

(One thing, STFU you VsCode Bonkers, There's more things other than VSCode lol)

**Reply by Reyororo** [reply]
I understand, but he's one of the people who helped me, so I won't do anything lol

**Reply by Reyororo** [reply]
Why don't you calm down a bit and make friends with me!

**Reply by Cheemsandfriends** [reply]
> Why don't you calm down a bit and make friends with me!

Well, At least I tried to help explaining my way lol.
But it makes me angry that ppl hate things just because someone doesn't use certain stuff or something.
Maybe I'm dumb idk xd
(And again, Use what you like! Obviously it's recommended to use Haxe-powered IDE's or something similar to make coding easy and simple, but Yeah lol)

**Reply by Reyororo** [reply]
> > Why don't you calm down a bit and make friends with me!
> 
> Well, At least I tried to help explaining my way lol. But it makes me angry that ppl hate things just because someone doesn't use certain stuff or something. Maybe I'm dumb idk xd (And again, Use what you like! Obviously it's recommended to use Haxe-powered IDE's or something similar to make coding easy and simple, but Yeah lol)

I think you're right. I think you're right, it's best to choose what you like, but he has his own way of thinking, so it's best to accept each other.
I'm looking forward to your response on the other thing about wanting to be friends with you:)

**Reply by Cheemsandfriends** [reply]
> 
> I think you're right. I think you're right, it's best to choose what you like, but he has his own way of thinking, so it's best to accept each other. I'm looking forward to your response on the other thing about wanting to be friends with you:)

Oh, Well, Yeah We can be friends I guess

**Reply by Reyororo** [reply]
> 

I'm glad to hear that So do you do discord?

**Reply by Cheemsandfriends** [reply]
> > 
> 
> I'm glad to hear that So do you do discord?

Wait What??? I don't do Discord 😳 lol.

**Reply by Reyororo** [reply]
> > > 
> > 
> > 
> > I'm glad to hear that So do you do discord?
> 
> Wait What??? I don't do Discord 😳 lol.

Okay... then do you have a Twitter account?

**Reply by Cheemsandfriends** [reply]
Yeah lol,
I have a Discord account and a Twitter one.

> > > > 
> > > 
> > > 
> > > I'm glad to hear that So do you do discord?
> > 
> > 
> > Wait What??? I don't do Discord 😳 lol.
> 
> Okay... then do you have a Twitter account?

Yes, I have a Discord account and Twitter account.
Just found funny about that you said that I do Discord lol

**Reply by Reyororo** [reply]
I'm not very good at English so my sentences are weird lol

**Reply by Reyororo** [reply]
> Yeah lol, I have a Discord account and a Twitter one.
> 
> > > > > 
> > > > 
> > > > 
> > > > I'm glad to hear that So do you do discord?
> > > 
> > > 
> > > Wait What??? I don't do Discord 😳 lol.
> > 
> > 
> > Okay... then do you have a Twitter account?
> 
> Yes, I have a Discord account and Twitter account. Just found funny about that you said that I do Discord lol

That's what I wanted to hear! I'd like to be friends with you on discord, if that's okay?

**Reply by Cheemsandfriends** [reply]
> > Yeah lol, I have a Discord account and a Twitter one.
> > > > > > 
> > > > > 
> > > > > 
> > > > > I'm glad to hear that So do you do discord?
> > > > 
> > > > 
> > > > Wait What??? I don't do Discord 😳 lol.
> > > 
> > > 
> > > Okay... then do you have a Twitter account?
> > 
> > 
> > Yes, I have a Discord account and Twitter account. Just found funny about that you said that I do Discord lol
> 
> That's what I wanted to hear! I'd like to be friends with you on discord, if that's okay?

Sure, I'd would like to pass my discord but, Not here.
You know, privacy stuff lol.
Please ask me on this [Twitter thing](https://twitter.com/Cheems15685374)

**Reply by Cheemsandfriends** [reply]
> I'm not very good at English so my sentences are weird lol

It's Ok, No one learns English when they're born
